pallets/pallet-dip-provider/src/lib.rs

```diff
@@ -158,30 +158,27 @@ pub mod pallet {
 				T::IdentityCommitmentGenerator::generate_commitment(&identifier, &identity, commitment_version)
 					.map_err(|error| Error::<T>::IdentityCommitmentGenerator(error.into()))?;
 
-			IdentityCommitments::<T>::try_mutate(&identifier, commitment_version, |commitment_entry| {
-				if let Some(old_commitment) = commitment_entry {
-					T::ProviderHooks::on_commitment_removed(
-						&identifier,
-						&dispatcher,
-						old_commitment,
-						commitment_version,
-					)
-					.map_err(|e| Error::<T>::Hook(e.into()))?;
-					Self::deposit_event(Event::<T>::VersionedIdentityDeleted {
-						identifier: identifier.clone(),
-						version: commitment_version,
-					});
+			match Self::delete_identity_commitment_storage_entry(&identifier, commitment_version) {
+				// If an old commitment is found, call the relevant hook to deal with it.
+				Ok(commitment) => {
+					T::ProviderHooks::on_commitment_removed(&identifier, &dispatcher, &commitment, commitment_version)
```

Comment:
I think this should be inside the `delete_identity_commitment_storage_entry` function

Reply:
The function is called from the storage deposit hook: https://github.com/KILTprotocol/kilt-node/blob/c90b64b577407f661dd2fe810dfa537d73d3009f/dip-template/runtimes/dip-provider/src/dip.rs#L133. The only way to add it in there (I tried before already) is to remove the `submitter` parameter from the hook. What do you think?

Reply:
The deposit hook does not have information about the `submitter`. I thought about using the deposit owner as the submitter, but that felt wrong.

Reply:
Mh you could also make the dispatcher optional. Having the deposit owner there might also work since they are the one who reclaimed the deposit right?

Reply:
Yes, that is an assumption we make in our code, and hopefully it holds all the time. I would go that way. Making the dispatcher optional now means that the deposit has been reclaimed, but not necessarily from someone. I will rename `submitter` to `owner` 👍

Reply:
https://github.com/KILTprotocol/kilt-node/pull/592/commits/ad3317a75146bd620e550dc5ffebf89dc0e3dd3d